in_toto/verifylib.py

```diff
@@ -239,16 +239,17 @@ def verify_all_steps_signatures(layout, chain_link_dict):
 
   """
   for step in layout.steps:
-    keys_dict = {}
-
     # Find the according link for this step
     key_link_dict = chain_link_dict[step.name]
 
-    # Create the dictionary of keys for this step
-    for pubkeyid in step.pubkeys:
-      keys_dict[pubkeyid] = layout.keys[pubkeyid]
-
     for keyid, link in six.iteritems(key_link_dict):
+      keys_dict = {}
+
+      # Create the dictionary of keys for this step
+      for pubkeyid in step.pubkeys:
+          if pubkeyid == keyid:
+            keys_dict[pubkeyid] = layout.keys[pubkeyid]
```

Comment:
> if pubkeyid in step.pubkeys:
  keys_dict[pubkeyid] = layout.keys[pubkeyid]
else:
  raise SignatureVerificationError("<some descriptive message here>")


@lukpueh  Do we need this? This will raise an error if the keyid (as in the filename) is not present in the layout. But if it isn't present in the layout, our initial load_metadata will fail.
Or maybe I'm missing something.